cluster/addons/dns/kube2sky/kube2sky_test.go

```diff
@@ -0,0 +1,189 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"encoding/json"
+	"path"
+	"strings"
+	"testing"
+	"time"
+
+	kapi "github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/coreos/go-etcd/etcd"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+type fakeEtcdClient struct {
+	writes map[string]string
+}
+
+func (ec *fakeEtcdClient) Set(path, value string, ttl uint64) (*etcd.Response, error) {
+	ec.writes[path] = value
+	return nil, nil
+}
+
+func (ec *fakeEtcdClient) Delete(path string, recursive bool) (*etcd.Response, error) {
+	for p, _ := range ec.writes {
+		if strings.HasPrefix(p, path) {
+			delete(ec.writes, p)
+		}
+	}
+	return nil, nil
+}
+
+const (
+	testDomain = "cluster.local"
+	basePath   = "/skydns/local/cluster"
+)
+
+func newKube2Sky() (*kube2sky, *fakeEtcdClient) {
```

Comment:
Not possible since the etcdClient in kube2sky object is an interface. I have done something different to make you happy :)